automation/src/main/java/org/wso2/carbon/testgrid/automation/file/TestNGTestReader.java

```diff
@@ -18,19 +18,57 @@
 
 package org.wso2.carbon.testgrid.automation.file;
 
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.testgrid.automation.beans.JMeterTest;
 import org.wso2.carbon.testgrid.automation.beans.Test;
+import org.wso2.carbon.testgrid.automation.beans.TestNGTest;
 import org.wso2.carbon.testgrid.automation.exceptions.TestReaderException;
 import org.wso2.carbon.testgrid.automation.file.common.TestReader;
+import org.wso2.carbon.testgrid.common.constants.TestGridConstants;
 
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
 /**
  * This class is responsible for reading testNG tests for TestGrid framework.
  */
 public class TestNGTestReader implements TestReader {
 
+    private static final Log log = LogFactory.getLog(JMeterTestReader.class);
+
+    /**
+     * This method goes through the file structure and create an object model of the tests.
+     *
+     * @param file File object for the test folder.
+     * @return a List of Test objects.
+     */
+    private List<Test> processTestStructure(File file) {
+        List<Test> testsList = new ArrayList<>();
+        File tests = new File(file.getAbsolutePath());
+        TestNGTest test = new TestNGTest();
```

Comment:
Better use `Test test = new TestNGTest()` (Programming for interface)

Reply:
There is a another method in TestNG and hence the interface cannot be used